alf/networks/preprocessors.py

```diff
@@ -129,3 +129,161 @@ def _preprocess(self, tensor):
         ret = self._embedding_net(tensor)
         # EncodingNetwork returns a pair
         return (ret if self._input_tensor_spec.is_discrete else ret[0])
+
+
+@gin.configurable
+class SharedEmbeddingPreprocessor(InputPreprocessor, SingletonInstanceNetwork):
```

Comment:
This inheritance is confusing. InputPreprocessor is not singleton while SingletonInstanceNetwork is. And mostly what it does is EmbeddingPreprocessor's thing. Is there any better way of constructing this class?

Reply:
I have simplified the code and where ```SharedEmbeddingPreprocessor``` is removed completely. Will push a commit soon.